src/modules/localization/src/frame_tf_calibrate/frame_calibration.cpp

```diff
@@ -0,0 +1,231 @@
+#include "localization/frame_tf_calibrate/frame_calibration.h"
+/*
+NOTE: This node listens to the tf broadcasts and performs the homogeneous transforms.
+      Another major function of this node is to "calibrate" the initial position and orientation
+      of the vehicle. The homogeneous transforms are applied and then the result is published.
+      
+INPUTS: TOPIC:  /gps/geodesy_odom
+                    Msg: nav_msgs::Odometry
+        TOPIC:  /localization/rotated_yaw
+                    Msg: std_msgs::Float32
+        TOPIC:  /localization/loam_odom_with_covar
+                    Msg: nav_msgs::Odometry
+
+        TF Listeners: Geodesy TF Message
+                      Lidar TF Message
+
+OUTPUTS: TOPIC: /localization/calibrated_pose
+                    Msg: geometry_msgs::Point
+         TOPIC: /localization/calibrated_yaw
+                    Msg: std_msgs::Float32
+         TOPIC: /localization/geodesy_tf
+                    Msg: nav_msgs::Odometry
+         TOPIC: /localization/lidar_tf
+                    Msg: nav_msgs::Odometry
+*/
+
+namespace fusionad
+{
+namespace localization
+{
+namespace frame_calibration_node
+{
+    FrameCalibrationNode::FrameCalibrationNode(){}
+
+    FrameCalibrationNode::~FrameCalibrationNode(){}
+
+    void FrameCalibrationNode::initRosComm()
+    {
+        // publisher for gps position calibration

```

Comment:
The two calibration values need to be consolidated into a custom message so it can be stored and published under one topic called "localization/calibration

Reply:
Okay, I'll just use a `geometry_msgs::Pose` message as shown [here](http://docs.ros.org/api/geometry_msgs/html/msg/Pose.html)

Reply:
What is the use of the calibrated position and yaw? Is it used as an offset value or is it used as an reference point in other files?

Reply:
The loam_velodyne package uses scan matching to calculate the vehicle's pose and orientation. This brings up a problem because that means it initializes at x = 0, y = 0, and yaw = 0. The purpose of ISS-141 was to feed an initial condition to frame_calibration.cpp and have this script make the appropriate homogeneous transform to go from the lidar's `base_link` frame to the `odom` frame.

Reply:
Ok, then ```geometry_msgs::Pose``` sounds good.